src/Workspaces/Core/Desktop/Workspace/Storage/StorageDatabaseLogger.cs

```diff
@@ -8,15 +8,22 @@ namespace Microsoft.CodeAnalysis.Storage
 {
     internal class StorageDatabaseLogger
     {
+        private static Exception s_reportedException;
```

Comment:
Having these all be statics actually makes them harder to find in crash dumps.  Making this a singleton instance with fields would aid debuggability.